packages/flutter_tools/lib/src/build_system/targets/darwin.dart

```diff
@@ -0,0 +1,147 @@
+      throw MissingDefineException(kBuildMode, name);
+    }
+    final BuildMode buildMode = BuildMode.fromCliName(buildModeEnvironment);
+    final String? devDependenciesEnabledString = environment.defines[kDevDependenciesEnabled];
+    if (devDependenciesEnabledString == null) {
+      throw MissingDefineException(kDevDependenciesEnabled, name);
+    }
+
+    final bool? devDependenciesEnabled = bool.tryParse(
+      environment.defines[kDevDependenciesEnabled] ?? '',
+    );
+    if (devDependenciesEnabled == null) {
+      throw Exception(
+        'Unexpected $kDevDependenciesEnabled define value: "$devDependenciesEnabledString"',
+      );
+    }
+
+    if (devDependenciesEnabled && buildMode.isRelease) {
+      // Supress this error if the project has no dev dependencies.
+      if (!_hasDevDependencies(environment)) {
+        environment.logger.printTrace(
+          'Ignoring dev dependencies error as the project has no dev dependencies',
+        );
+        return;
+      }
+
+      _printXcodeError(
+        'Release builds should not have Dart dev dependencies enabled\n'
+        '\n'
+        'This error happens if:\n'
+        '\n'
+        '1. Your pubspec.yaml has dev dependencies\n'
+        '2. Your last Flutter CLI action turned on dev dependencies\n'
+        '3. You do a release build in Xcode\n'
+        '\n'
+        'You can turn off Dart dev dependencies by running this in your Flutter project:\n'
+        '\n'
+        '  $releaseBuildCommand\n'
+        '\n',
+      );
+      throwToolExit('Dev dependencies enabled in release build');
+    } else if (!devDependenciesEnabled && !buildMode.isRelease) {
+      // Supress this error if the project has no dev dependencies.
+      if (!_hasDevDependencies(environment)) {
+        environment.logger.printTrace(
+          'Ignoring dev dependencies error as the project has no dev dependencies',
+        );
+        return;
+      }
+
+      final bool profile = buildMode == BuildMode.profile;
+      _printXcodeError(
+        '${profile ? 'Profile' : 'Debug'} builds require Dart dev dependencies\n'
+        '\n'
+        'This error happens if:\n'
+        '\n'
+        '1. Your pubspec.yaml has dev dependencies\n'
+        '2. Your last Flutter CLI action turned off dev dependencies\n'
+        '3. You do a debug or profile build in Xcode\n'
+        '\n'
+        'You can turn on Dart dev dependencies by running this in your Flutter project:\n'
+        '\n'
+        '  $debugBuildCommand\n'
+        '\n',
+      );
+      throwToolExit('Dev dependencies disabled in ${profile ? 'profile' : 'debug'} build');
+    }
+  }
+
+  // Check if the iOS project has Dart dev dependencies. On error, assumes true.
+  bool _hasDevDependencies(Environment environment) {
+    // If the app has no plugins, the .flutter-plugins-dependencies file isn't generated.
+    final File pluginsFile = environment.projectDir.childFile('.flutter-plugins-dependencies');
```

Comment:
I'd avoid using a hardcoded path, since it could change. 

```suggestion
    final FlutterProject flutterProject = FlutterProject.fromDirectory(environment.projectDir);
    final File pluginsFile = flutterProject.flutterPluginsDependenciesFile;
```

Reply:
Updated!